assets/js/app/llms-forms.js

```diff
@@ -0,0 +1,430 @@
+
+	/**
+	 * Bind DOM Events fields with dynamic localization values and language.
+	 *
+	 * @since [version]
+	 *
+	 * @return {void}
+	 */
+	bind_l10n_selects: function() {
+
+		var self = this;
+
+		self.$countries = $( '.llms-l10n-country-select select' );
+		self.$states    = $( '.llms-l10n-state-select select' );
+		self.$zips      = $( '#llms_billing_zip' );
+
+		if ( ! self.$countries.length ) {
+			return;
+		}
+
+		if ( self.$states.length ) {
+			self.prep_state_field();
+		}
+
+		self.$countries.add( self.$states ).llmsSelect2( { width: '100%' } );
+
+		if ( window.llms.locale ) {
+			self.locale = JSON.parse( window.llms.locale );
+			self.locale_defaults = {
+				state: ( function() {
+					return self.get_label_text( self.get_field_parent( self.$states ).find( 'label' ) );
+				} )(),
+			};
+		}
+
+		self.$countries.on( 'change', function() {
+
+			var val = $( this ).val();
+			self.update_state_options( val );
+			self.update_locale_info( val );
+
+		} ).trigger( 'change' );
+
+	},
+
+	/**
+	 * Ensure "matching" fields match.
+	 *
+	 * @since [version]
+	 *
+	 * @return {Void}
+	 */
+	bind_matching_fields: function() {
+
+		var $fields = $( 'input[data-match]' ).not( '[type="password"]' );
+
+		$fields.each( function() {
+
+			var $field = $( this ),
+				$match = $( '#' + $field.attr( 'data-match' ) ),
+				$parents;
+
+			if ( $match.length ) {
+
+				$parents = $field.closest( '.llms-form-field' ).add( $match.closest( '.llms-form-field' ) );
+
+				$field.on( 'input change', function() {
+
+					var val_1 = $field.val(),
+						val_2 = $match.val();
+
+					if ( val_1 && val_2 && val_1 !== val_2 ) {
+						$parents.addClass( 'invalid' );
+					} else {
+						$parents.removeClass( 'invalid' );
+					}
+
+				} );
+
+			}
+
+		} );
+
+	},
+
+	/**
+	 * Bind DOM events for voucher toggles UX.
+	 *
+	 * @since [version]
+	 *
+	 * @return {void}
+	 */
+	bind_voucher_field: function() {
+
+		$( '#llms-voucher-toggle' ).on( 'click', function( e ) {
+			e.preventDefault();
+			$( '#llms_voucher' ).toggle();
+		} );
+
+	},
+
+	/**
+	 * Retrieve the parent element for a given field.
+	 *
+	 * The parent element is hidden when the field isn't required.
+	 * Looks for a WP column wrapper and falls back to the field's
+	 * wrapper div.
+	 *
+	 * @since [version]
+	 *
+	 * @see {Reference}
+	 * @link {URL}
+	 *
+	 * @param {[type]} $field
+	 * @return {[type]}
+	 */
+	get_field_parent: function( $field ) {
+
+		var $block = $field.closest( '.wp-block-column' );
+		if ( $block.length ) {
+			return $block;
+		}
+
+		return $field.closest( '.llms-form-field' );
+
+	},
+
+	/**
+	 * Retrieve the text of a label
+	 *
+	 * Removes an children HTML elements (eg: required span elemeents) and returns only the labels text.
+	 *
+	 * @since [version]
+	 *
+	 * @param {Object} $label jQuery object for a label element.
+	 * @return {String}
+	 */
+	get_label_text: function( $label ) {
+
+		var $clone = $label.clone();
+		$clone.find( '*' ).remove();
```

Comment:
I do believe that you're right however I looked it up, of course, and found the following: https://stackoverflow.com/a/648014/400568

This is just one guy on SO stating that it doesn't really matter and I'm going to side with him because I don't feel like retesting the code.

If you feel real strong about this I will rewrite and retest.

Reply:
Not worthwhile, don't do it.
I'm going to side with you who side with him :).